### PR TITLE
fix some implicit declarations

### PR DESCRIPTION
**Pull request by nekopsykose**
fixes implicit-function-declaration of strcasecmp (strings.h) and implicit define of struct timezone (bsd/gnu source)

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -11,6 +11,8 @@ m4_pattern_forbid([^AX_])
 
 AC_PROG_CC
 
+AC_DEFINE(_BSD_SOURCE, 1, [Required for struct timezone])
```
Comment (koutcher) [inline configure.ac:14]: Thanks. As configure is not the only way to compile Tig, I'd rather have the define done in include/tig/tig.h:37.

Comment (nekopsykose) [reply configure.ac]: updated, right below the config.h include

Comment (koutcher) [reply configure.ac]: Thanks. Looks like we also need to add an extra `#define _DEFAULT_SOURCE` for glibc >= 2.19.

Comment (nekopsykose) [reply configure.ac]: ah, right

**Pull request merged**